cfme/ansible/credentials.py

```diff
@@ -155,81 +157,22 @@ def before_fill(self, values):
-            "privilage_escalation": credentials.get("privilage_escalation"),
-            "privilage_escalation_username": credentials.get("privilage_escalation_username"),
-            "privilage_escalation_password": credentials.get("privilage_escalation_password"),
-            "vault_password": credentials.get("vault_password")
-        }
-        scm_credential_fill_dict = {
-            "username": credentials.get("username"),
-            "password": credentials.get("password"),
-            "private_key": credentials.get("private_key"),
-            "private_key_phrase": credentials.get("private_key_phrase")
-        }
-        amazon_credential_fill_dict = {
-            "access_key": credentials.get("access_key"),
-            "secret_key": credentials.get("secret_key"),
-            "sts_token": credentials.get("sts_token"),
-        }
-        vmware_credential_fill_dict = {
-            "username": credentials.get("username"),
-            "password": credentials.get("password"),
-            "vcenter_host": credentials.get("vcenter_host")
-        }
-        credential_type_map = {
-            "Machine": machine_credential_fill_dict,
-            "Scm": scm_credential_fill_dict,
-            "Amazon": amazon_credential_fill_dict,
-            "VMware": vmware_credential_fill_dict
-        }
-
-        add_page.fill({"name": name, "credential_type": credential_type})
-        add_page.credential_form.fill(credential_type_map[credential_type])
-        add_page.add_button.click()
-        credentials_list_page = self.create_view(CredentialsListView)
-        # Without this StaleElementReferenceException can be raised
-        wait_for(lambda: False, silent_failure=True, timeout=5)
-        assert credentials_list_page.is_displayed
-        credentials_list_page.flash.assert_success_message(
-            'Add of Credential "{}" has been successfully queued.'.format(name))
-
-        credential = self.instantiate(name, credential_type, **credentials)
-
-        wait_for(
-            lambda: credential.exists,
-            fail_func=credentials_list_page.browser.selenium.refresh,
-            delay=5,
-            timeout=300)
-
-        return credential
-
-
 class Credential(BaseEntity):
     """A class representing one Embedded Ansible credential in the UI."""
+
+    # TODO - This is one of the only classes that hasn't been converted to attrs
+    # The class needs to be reworked and split into multiple subtypes. The kwargs
+    # is also problematic for attrs
+
     def __init__(self, collection, name, credential_type, **credentials):
-        self.collection = collection
-        self.appliance = self.collection.appliance
+        super(Credential, self).__init__(collection)
         self.name = name
         self.credential_type = credential_type
         for key, value in credentials.iteritems():
             setattr(self, key, value)
 
+    __repr__ = object.__repr__
```

Comment:
We discussed this on the phone, but I'd like to see an in-line comment as to why we have to set _repr_ here - since we won't once we have specific credential classes.